kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -84,28 +84,35 @@ class KyuubiSessionImpl(
     super.open()
     // Init zookeeper client here to capture errors
     zkClient
-    getServerHost match {
-      case Some((host, port)) => openSession(host, port)
-      case None =>
-        configureSession()
-        val builder = new SparkProcessBuilder(user, sessionConf.toSparkPrefixedConf)
-        val process = builder.start
-        info(s"Launching SQL engine: $builder")
-        var sh = getServerHost
-        val started = System.currentTimeMillis()
-        while (sh.isEmpty) {
-          if (process.waitFor(1, TimeUnit.SECONDS)) {
-            throw builder.getError
-          }
-          if (started + timeout <= System.currentTimeMillis()) {
-            process.destroyForcibly()
-            throw KyuubiSQLException(s"Timed out($timeout ms) to launched Spark with $builder",
-              builder.getError)
+    try {
```

Comment:
Seems that we don't need this try-catch here as all are handled by the session manager.

How about just add this to the end?

```scala
try {
   zkClient.close()
} catch {
   case e: IOException => error("Failed to release the zkClient after session established", e)
}
```

Reply:
How about all kyuubi's sessions share one ZkClient?

Reply:
I don‘t trust zk’s stability very much :)
For registering server or engines, we have a de-register handler there.

Reply:
here, I don't think it is a right way to go

Reply:
i see